terraform/providers/vsphereprivate/resource_vsphereprivate_tag_attach.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func resourceVSpherePrivateTagAttach() *schema.Resource {
+	return &schema.Resource{
+		Create:        resourceVSpherePrivateTagAttachCreate,
+		Read:          resourceVSpherePrivateTagAttachRead,
+		Delete:        resourceVSpherePrivateTagAttachDelete,
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"objectid": {
+				Type:         schema.TypeString,
+				Description:  "The managed object id of the object to attach the tag to.",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"objecttype": {
+				Type:         schema.TypeString,
+				Description:  "The type of the managed object.",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"tagid": {
+				Type:         schema.TypeString,
+				Description:  "The id of tag.",
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.NoZeroValues,
+			},
+		},
+	}
+}
+
+func resourceVSpherePrivateTagAttachCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Print("[DEBUG] beginning of tag attachment")
+
+	ctx, cancel := context.WithTimeout(context.TODO(), defaultAPITimeout)
+	defer cancel()
+	tagManager := tags.NewManager(meta.(*VSphereClient).restClient)
+	moRef := types.ManagedObjectReference{
+		Value: d.Get("objectid").(string),
+		Type:  d.Get("objecttype").(string),
+	}
+
+	err := tagManager.AttachTag(ctx, d.Get("tagid").(string), moRef)
+
+	if err != nil {
+		return fmt.Errorf("error managed object: %s not found, managed object id: %s", d.Get("objectid"), d.Id())
```

Comment:
Do we want to also bubble up the error returned by tagManager.AttachTag()? It might contained more detailed information on the type of error.

Reply:
@sadasu is that what you were thinking?